iroh-net/src/endpoint.rs

```diff
@@ -87,117 +104,84 @@ impl Default for Builder {
 }
 
 impl Builder {
-    /// Set a secret key to authenticate with other peers.
+    /// Sets a secret key to authenticate with other peers.
     ///
-    /// This secret key's public key will be the [PublicKey] of this endpoint.
+    /// This secret key's public key will be the [`PublicKey`] of this endpoint and thus
+    /// also its [`NodeId`][]
     ///
     /// If not set, a new secret key will be generated.
     pub fn secret_key(mut self, secret_key: SecretKey) -> Self {
         self.secret_key = Some(secret_key);
         self
     }
 
-    /// Set the ALPN protocols that this endpoint will accept on incoming connections.
-    pub fn alpns(mut self, alpn_protocols: Vec<Vec<u8>>) -> Self {
-        self.alpn_protocols = alpn_protocols;
-        self
-    }
-
-    /// Set an explicit proxy url to proxy all HTTP(S) traffic through.
-    pub fn proxy_url(mut self, url: Url) -> Self {
-        self.proxy_url.replace(url);
-        self
-    }
-
-    /// Set the proxy url from the environment, in this order:
+    /// Set the [ALPN] protocols that this endpoint will accept on incoming connections.
```

Comment:
I don't personally care, but in order to keep this in line with the intended result...
```suggestion
    /// Sets the [ALPN] protocols that this endpoint will accept on incoming connections.
```